test/TestCase.cpp

```diff
@@ -188,18 +192,23 @@ bool EVMVersionRestrictedTestCase::validateSettings(langutil::EVMVersion _evmVer
 	if (!version)
 		BOOST_THROW_EXCEPTION(runtime_error{"Invalid EVM version: \"" + versionString + "\""});
 
+	langutil::EVMVersion evmVersion = solidity::test::CommonOptions::get().evmVersion();
```

Comment:
Is it a good thing to get this setting from the singleton? Would it be possible to rather "inject" the setting instead?

Reply:
Basically add a function `implicitSetting()` that adds something to the settings map without reading it from the file itself?

Reply:
Ah wait a second, I'm totally mistaken. This is the environment we are running in, not the setting. So I think in that case it might be fine.